### PR TITLE
TandemFoil: lr=1.25e-4 + gc=1.0 — deeper LR sweep with gradient clipping

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
lr=1.5e-4 just beat lr=2e-4 (44.72 vs 45.07, PR #2724). The LR trend is: 3e-4→2e-4→1.5e-4 all improving. We test lr=1.25e-4 (continuing the sweep) combined with gc=1.0 (which reduced destructive cosine-peak spikes observed at lr=1.5e-4 — spikes to 80-90 at cosine peaks). The combination may stabilize optimization and allow the trough envelope to descend further.

## Instructions
\`\`\`bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 1.25e-4 \
  --cosine-t-max 10 \
  --grad-clip 1.0 \
  --no-use-ema \
  --model-slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --agent gilbert \
  --wandb-name "gilbert/tandem-lr125e4-gc10" \
  --wandb-group "tandem-lr-gc-sweep"
\`\`\`

Report val_primary/surface_pressure_mae at best epoch, total epochs, time per epoch, and whether oscillation amplitude decreased vs lr=1.5e-4.

## Baseline
Current best: **44.72** at epoch 89 (3L/192d, lr=**1.5e-4**, T_max=10, NO grad-clip) — PR #2724
Trend: lr=3e-4(52.81)→lr=2e-4(45.07)→lr=1.5e-4(44.72) — lower LR keeps winning

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT gilbert:

## Results

### Summary
**lr=1.25e-4 + gc=1.0 is a massive win — 32.7% improvement over baseline.**

| Metric | Baseline (lr=1.5e-4, no gc) | This run (lr=1.25e-4, gc=1.0) | Change |
|--------|---------------------------|-------------------------------|--------|
| **val_primary/surface_pressure_mae** | 44.72 (ep89) | **30.10** (ep157) | **-32.7%** |
| test_primary/surface_pressure_mae | — | 33.88 | — |

### Per-split val at best (ep157)
| Split | MAE |
|-------|-----|
| geom_camber_cruise | 17.12 |
| re_rand | 28.63 |
| single_in_dist | 32.68 |
| geom_camber_rc | 41.95 |

### Per-split test
| Split | MAE |
|-------|-----|
| re_rand | 25.73 |
| geom_camber_cruise | 34.34 |
| single_in_dist | 35.11 |
| geom_camber_rc | 40.35 |

### Oscillation analysis — gc=1.0 is transformative
The grad clipping dramatically changes the optimization dynamics:
- **Baseline (lr=1.5e-4, no gc):** Spikes to 80-90 at cosine peaks, 20-30 point oscillation amplitude
- **This run (lr=1.25e-4, gc=1.0):** Max peak bounce ~7 points, typical ~2-3 points, often <1 point
- Grad clip events dropped from 750/750 (ep1) → 15-20/750 (late epochs), meaning gradients are naturally well-behaved after initial training
- The combination allows the model to train much longer without destructive spikes

### Training dynamics
- **162 epochs** in 180-min budget (~1.1 min/epoch) vs baseline's 115 epochs (~1.57 min/epoch)
- Faster epoch time likely from grad clipping reducing gradient explosion overhead
- **Still improving at timeout** — trough envelope descended from ~164 (ep1) → 42 (ep37-42 range) → 31 (ep145-157) with no sign of convergence
- Three distinct descent phases: rapid (ep1-30), oscillating descent (ep30-90), deep basin exploration (ep90-162)
- Loss continued decreasing to 0.055 at ep143 — lowest observed

### Command
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 1.25e-4 \
  --cosine-t-max 10 \
  --grad-clip 1.0 \
  --no-use-ema \
  --model-slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --agent gilbert \
  --wandb-name "gilbert/tandem-lr125e4-gc10" \
  --wandb-group "tandem-lr-gc-sweep"
```

### W&B run
- **Run ID:** v6amjkh7
- **URL:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/v6amjkh7

### What happened
The lr=1.25e-4 + gc=1.0 combination is a clear winner. The key insight is that gradient clipping doesn't just reduce oscillation — it fundamentally changes the optimization landscape by preventing destructive gradient explosions at cosine LR peaks. This allows the model to:
1. Train longer (162 vs 115 epochs in the same budget)
2. Explore deeper basins without being knocked out by gradient spikes
3. Maintain stable descent far beyond where the baseline plateaued

The LR trend continues: 3e-4(52.81) → 2e-4(45.07) → 1.5e-4(44.72) → **1.25e-4+gc(30.10)**. But this jump is 10x larger than previous gains, suggesting gc=1.0 is the dominant lever, not just the LR reduction.

### Suggested follow-ups
1. **lr=1.25e-4 without gc** — isolate whether the improvement comes from LR alone or the gc+LR synergy
2. **lr=1e-4 + gc=1.0** — continue the LR sweep with gc now as a fixed component
3. **lr=1.25e-4 + gc=0.5** — try tighter clipping since grad norms are already low (0.5 by late training)
4. **Longer training budget** — model was still improving at 162 epochs; 300-min run could push below 30
5. **Cross-dataset test** — gc=1.0 may benefit AirfRANS and DrivAerML similarly
